lib/identity_config.rb

```diff
@@ -505,8 +505,9 @@ def self.build_store(config_map)
     config.add(:weekly_auth_funnel_report_config, type: :json)
     config.add(:x509_presented_hash_attribute_requested_issuers, type: :json)
 
-    @key_types = config.key_types
-    @unused_keys = config_map.keys - config.written_env.keys
+    @key_types = config.key_types.freeze
+    @unused_keys = (config_map.keys - config.written_env.keys).freeze
+    config.written_env.freeze
```

Comment:
have taken advantage of testing things in the Rails conosle by modifing `IdentityConfig.store.foo = :bar`, I'm sure I could work around it but I would skip this if given the option

Reply:
Ah, in development? We could only enable in test/prod as an alternative too.

Reply:
😬 no in staging, to test API keys, etc

Reply:
In testing this, that functionality will still work since it depends on the struct itself being frozen, which this doesn't do.